service/k8sclient/k8sclient_test.go

```diff
@@ -373,3 +378,152 @@ func assertListPSMDBCluster(ctx context.Context, t *testing.T, client *K8sClient
 		}
 	}
 }
+
+func TestConvertToCPUMilis(t *testing.T) {
+	t.Parallel()
+	testCases := []struct {
+		in             string
+		expectedOut    int64
+		errShouldBeNil bool
+	}{
+		{in: "100m", expectedOut: 100, errShouldBeNil: true},
+		{in: "1", expectedOut: 1000, errShouldBeNil: true},
+		{in: "1.252", expectedOut: 1252, errShouldBeNil: true},
+		{in: "0.252", expectedOut: 252, errShouldBeNil: true},
+		{in: "0.0", expectedOut: 0, errShouldBeNil: true},
+		{in: "0.", expectedOut: 0, errShouldBeNil: false},
+		{in: ".0", expectedOut: 0, errShouldBeNil: false},
+		{in: ".", expectedOut: 0, errShouldBeNil: false},
+		{in: "", expectedOut: 0, errShouldBeNil: false},
+		{in: "adf", expectedOut: 0, errShouldBeNil: false},
+	}
+
+	for _, test := range testCases {
+		out, err := convertToCPUMilis(test.in)
+		assert.Equal(t, test.expectedOut, out, "in=%v, out=%v, err=%v", test.in, out, err)
+		assert.Equal(
+			t, test.errShouldBeNil, err == nil,
+			"in=%v, out=%v, errShouldBeNil=%v: actually err == nil is %v\nerr=%v",
+			test.in, out, test.errShouldBeNil, err == nil, err,
+		)
+	}
+}
+
+func TestConvertToBytes(t *testing.T) {
+	t.Parallel()
+	testCases := []struct {
+		in             string
+		expectedOut    int64
+		errShouldBeNil bool
+	}{
+		{in: "100M", expectedOut: 100 * 1000 * 1000, errShouldBeNil: true},
+		{in: "100Mi", expectedOut: 100 * 1024 * 1024, errShouldBeNil: true},
+		{in: "100", expectedOut: 100, errShouldBeNil: true},
+		{in: "1G", expectedOut: 1000 * 1000 * 1000, errShouldBeNil: true},
+		{in: "1Gi", expectedOut: 1024 * 1024 * 1024, errShouldBeNil: true},
+		{in: "0.5Gi", expectedOut: 1024 * 1024 * 1024 / 2, errShouldBeNil: true},
+		{in: "0.3Gi", expectedOut: 322122548, errShouldBeNil: true},
+		{in: "3000m", expectedOut: 3, errShouldBeNil: true},
+		{in: "Gi", expectedOut: 0, errShouldBeNil: false},
+		{in: "", expectedOut: 0, errShouldBeNil: false},
+		{in: "1Z", expectedOut: 0, errShouldBeNil: false},
+		{in: "1Ki", expectedOut: 1024, errShouldBeNil: true},
+		{in: "1K", expectedOut: 1000, errShouldBeNil: true},
+		{in: "1T", expectedOut: 1000 * 1000 * 1000 * 1000, errShouldBeNil: true},
+		{in: "1Ti", expectedOut: 1024 * 1024 * 1024 * 1024, errShouldBeNil: true},
+	}
+
+	for _, test := range testCases {
+		out, err := convertToBytes(test.in)
+		assert.Equal(t, test.expectedOut, out, "in=%v, out=%v, err=%v", test.in, out, err)
+		assert.Equal(
+			t, test.errShouldBeNil, err == nil,
+			"in=%v, out=%v, errShouldBeNil=%v: actually err == nil is %v\nerr=%v",
+			test.in, out, test.errShouldBeNil, err == nil, err,
+		)
+	}
+}
+
+func TestGetConsumedResources(t *testing.T) {
+	t.Parallel()
+	ctx := app.Context()
+
+	kubeconfig, err := ioutil.ReadFile(os.Getenv("HOME") + "/.kube/config")
```

Comment:
For other tests we use different approach to get kube config which should work better.

Reply:
I did not work with EKS tests so I changed it. It works for minikube because all fields are present in output when you call 
https://github.com/percona-platform/dbaas-controller/blob/main/service/k8sclient/k8sclient_test.go#L49, but for EKS, the config contains certs that are trimmed out of the output for security reasons I guess.

Reply:
oh, ok then.